### DISCUSSION TITLE
Queue system when importing

### DISCUSSION THREAD

**Discussion by CoenWarmer** [directus/directus#14204] [Feature Requests]
We often make exports of items with a lot of nested items. This can in some cases lead to JSON files that are in excess of 300kb.

When we then import these again using POST on /api/items/<collectionType>, this can in some cases lead to processing times of over 60 seconds. 

This is not a problem (performance is not the issue here, it is fine that this takes a while) but it becomes a bit problematic because a part of our infra cuts off requests that take over 60 seconds because it assumes the underlying service has timed out.

It would be really cool if Directus would support some sort of queuing mechanism where it would take a payload, upon successful placement in the queue return a 200, then support an endpoint where it would be possible to poll for the status of the queue / queue item, or upon handling of the queue send an SSE. What do you think?

**Comment by CoenWarmer** [comment]
Just noticed: 
* https://github.com/directus/directus/discussions/13299

**Comment by rijkvanzanten** [comment]
Fully agreed. Exports already allow exporting to directus_files which does the processing "async" under the hood. We should have the same for imports 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
